### PR TITLE
Handle quotes in command-line options

### PR DESCRIPTION
**Pull request by SetTrend**
Added support for quotes around file names from command line.

### CODE REVIEW

File: lib/tsc.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 /*! *****************************************************************************
-Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (mhegazy) [inline lib/tsc.js]: Do not change this file manually. this is a generated file from the build, updated by running Jake LKG.

File: src/compiler/commandLineParser.ts [hunk 2]
```diff
@@ -364,6 +364,7 @@ namespace ts {
 
         function parseStrings(args: string[]) {
             let i = 0;
+            function adjustFileName(arg: string): string { /*jshint -W110 */ return (arg[0] === "'" || arg[0] === '"') && arg[0] === arg[arg.length - 1] ? arg.substring(1, arg.length - 1) : arg; }
```
Comment (mhegazy) [inline src/compiler/commandLineParser.ts]: split on multiple lines, move outside the closure, and remove the jshint comment

Comment (mhegazy) [inline src/compiler/commandLineParser.ts]: may be rename to remove quoats

Comment (mhegazy) [inline src/compiler/commandLineParser.ts]: also, i would only handle double quotes and not single. i do not think that is a common or expected support anyways.

File: typings/jake.d.ts [hunk 3]
```diff
@@ -0,0 +1,414 @@
+// Type definitions for jake
```
Comment (mhegazy) [inline typings/jake.d.ts]: we should make the jakefile a .ts first anyways.

**Comment by mhegazy** [comment]
For future references, please file an issue first.

**Comment by DanielRosenwasser** [comment]
Work here shouldn't be done on the command line parser because the command line parser expects all escaping to be done by the shell. So work here should be restricted to the Jakefile.

As an aside, why do you need to do this exactly? Apart from tests, we usually [don't accept new files](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines) into the code base.

**Comment by DanielRosenwasser** [comment]
I'm going to close this in the interest of maintainability. Can you refile with no new `.d.ts` files and keeping things restricted to the Jakefile?

**Comment by SetTrend** [comment]
@DanielRosenwasser :

Please go ahead. You are right. I now realize that my changes didn't solve the problem I was trying to solve.

This was my first try of committing to TypeScript today. I learned a number of new insights which I haven't been aware of before. **I committed too fast. I'm very sorry for that.**

I'll re-create my fork and start over again.
<br/>
Actually, I'm hunting down the reason for `jake tests` failing, while the same command, when run directly through `TSC`, succeeds:

error

Trying to get `jake tests` to run successfully I saw the above `jake tests` errors and assumed it was caused by TSC because they are TSC errors. I then created a pull request to go through the full workflow and see what happens. I did this too early, without even making sure that `jake tests` runs successfully on my own machine.

I just returned to my desk now and I just noticed that the original TSC command doesn't fail, apparently even without my changes, while `jake tests` fails. I'll have a look at it again ...

File: lib/tsc.js (hunk 1 above)
Comment (SetTrend) [inline lib/tsc.js]: @mhegazy :

Out of curiosity: Shouldn't the `lib` folder be omitted from being maintained by git anyway? Wouldn't it be reasonable to add `lib` to `.gitignore` and have the Travis CI generate npm/NuGet packages from its local build?

**Pull request closed without merging**